### PR TITLE
Non-FD examples

### PR DESCRIPTION
**Pull request by FabioLuporini**
Some trivial examples showing how to express basic linear algebra operations

### CODE REVIEW

File: examples/misc/linalg.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+    """``Ax = b``."""
+    op = Operator(Eq(b, A*x), dle=optimize)
+    op.apply()
+    info('Executed `Ax = b`')
+
+
+def transpose(A, optimize):
+    """``A -> A^T``."""
+    i, j = A.indices
+    op = Operator(Eq(A, A.indexed[j, i]), dle=optimize)
```
Comment (mloubout) [inline examples/misc/linalg.py]: Don't think this works. In place transpose of a matrix are really tricky.
A[i, j] = A[j,.i]

so when reaches A[j, i] can't replace by A[i, j] anymore.

Can be tested with `np.dot(mat_vec(A,x), y) == np.dot(mat_vec(A^T,y),x)`

File: examples/misc/linalg.py [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    op.apply()
+    info('Executed `A -> A^T`')
+
+
+def transpose_mat_vec(A, x, b, optimize):
+    """``A -> A^T, A^Tx = b``."""
+    i, j = A.indices
+    op = Operator([Eq(A, A.indexed[j, i]), Eq(b, A*x)], dle=optimize)
```
Comment (mloubout) [inline examples/misc/linalg.py]: Same as above.
This one would work simply as `Eq(b, A.indexed[j, i]*x)`

File: examples/misc/linalg.py (hunk 1 above)
Comment (FabioLuporini) [reply examples/misc/linalg.py]: dumb me

File: examples/misc/linalg.py (hunk 2 above)
Comment (FabioLuporini) [reply examples/misc/linalg.py]: 👍

**Comment by FabioLuporini** [comment]
@mloubout can you take a final look? I addressed your comments (dumb me) and added one final example

**Comment by mloubout** [comment]
Looks good. Maybe for the future add some tests and run the examples on non-zeros.

**Comment by FabioLuporini** [comment]
Merged.

**Pull request merged**